App/features/photos/sagas.ts

```diff
@@ -22,16 +23,49 @@ function * toggleStorage(action: ActionType<typeof PreferencesActions.toggleStor
   }
 }
 
-function * nodeOnline() {
-  while (yield take([getType(TextileEventsActions.nodeOnline), getType(TextileEventsActions.stopNodeAfterDelayCancelled)])) {
+function * bootstrapPhotoProcessing() {
+  const QUEUE_CONCURRENT = 1
+  const { watcher, addTaskChannel } = yield createQueue(photoHandler, QUEUE_CONCURRENT)
+  yield fork(watcher)
+  yield call(processPending, addTaskChannel)
+  yield fork(queryForNewPhotos, addTaskChannel)
+}
+
+function * photoHandler(payload: { id: string }) {
```

Comment:
This is our pipeline for processing a single photo. It pauses until the node is online and then prepares, adds, and cleans up the photo.